packages/astro/src/@types/astro.ts

```diff
@@ -1493,6 +1493,20 @@ export interface AstroUserConfig {
 			 */
 			prefixDefaultLocale: boolean;
 
+			/**
+			 * @docs
+			 * @name i18n.routing.redirectToDefaultLocale
+			 * @kind h4
+			 * @type {boolean}
+			 * @default `true`
+			 * @version 4.2.0
+			 * @description
+			 *
+			 * When `false`, Astro doesn't do a redirect from `/` to `/<defaultLocale>`. This option comes into play only
+			 * when `prefixDefaultLocale` is `true`.
```

Comment:
``````suggestion
			 * Configures whether or not the home URL (`/`) generated by `src/pages/astro.index`
			 * will redirect to `/[defaultLocale]` when `prefixDefaultLocale: true` is set.
			 * 
			 * Set `redirectToDefaultLocale: false` to disable this automatic redirection at the root of your site:
			 * ```js
			 * // astro.config.mjs
			 * export default defineConfig({
			 *   i18n:{
			 *     defaultLocale: "en",
			 * 		locales: ["en", "fr"],
			 *     routing: {
			 *       prefixDefaultLocale: true,
			 *       redirectToDefaultLocale: false
			 *     }
			 *   }
			 * })
			 *```
``````
We could include a code sample of the config here, too maybe?